libnetwork/iptables/firewalld.go

```diff
@@ -52,11 +60,54 @@ func firewalldInit() error {
 	if connection, err = newConnection(); err != nil {
 		return fmt.Errorf("Failed to connect to D-Bus system bus: %v", err)
 	}
-	firewalldRunning = checkRunning()
+
+	ictx, icancel := context.WithTimeout(context.Background(), dbusConnectionTimeout)
+	defer icancel()
+
+	firewalldRunning, err = checkRunning(ictx)
```

Comment:
```suggestion
	firewalldReady, err = checkRunning(ictx)
```
`checkRunning()` signals that firewalld is not running by returning a non-nil err. The boolean value returned will be true if-and-only-if firewalld is ready for us to configure.